automation/deploy_igz/deploy_remote.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python3
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import io
+import logging
+import os
+import re
+import subprocess
+import sys
+from typing import List
+
+import click
+import coloredlogs
+import paramiko
+import yaml
+
+log_level = logging.INFO
+fmt = "%(asctime)s %(levelname)s %(message)s"
+logging.basicConfig(level=log_level)
+logger = logging.getLogger("mlrun-deploy")
+coloredlogs.install(level=log_level, logger=logger, fmt=fmt)
+
+
+class MlRunDeployer(object):
+    class Consts(object):
+        mandatory_fields = ["DATA_NODES", "USER", "PASSWORD", "DOCKER_REGISTRY"]
+
+    def __init__(self, conf_file):
+        self._config = yaml.safe_load(conf_file)
+        for key in self.Consts.mandatory_fields:
+            if self._config.get(key, None) is None:
+                raise RuntimeError(f"Mandatory option {key} not defined")
+        self._username = self._config.get("REGISTRY_USERNAME")
+        self._password = self._config.get("REGISTRY_PASSWORD")
+        if self._username is not None and self._password is None:
+            raise RuntimeError(
+                "REGISTRY_USERNAME defined, yet REGISTRY_PASSWORD is not defined"
+            )
```

Comment:
I'd move all the validations to a private `_validate_config` method.
Also, not a must, but if you save the mandatory_fields as a set:
```python
mandatory_fields = {"DATA_NODES", "USER", "PASSWORD", "DOCKER_REGISTRY"}
```
You can do the following:
```python
missing_fields = mandatory_fields - set(self._config.keys())
if len(missing_fields) > 0:
    raise raise RuntimeError(f"Mandatory options not defined: {missing_fields}")
```